nodeadm/internal/api/bridge/decode.go

```diff
@@ -28,6 +28,9 @@ func DecodeNodeConfig(data []byte) (*internalapi.NodeConfig, error) {
 	if gvk.Group != api.GroupName {
 		return nil, fmt.Errorf("failed to decode %q, unexpected group: %s", gvk.Kind, gvk.Group)
 	}
+	if err := ValidateExternalType(obj, *gvk); err != nil {
```

Comment:
based on tests, this is breaking how we merge partial configs 😞
will we have to do the merging in json/yaml like kubelet? https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/server.go#L314-L345